lib/pbench/cli/server/report.py

```diff
@@ -343,8 +414,9 @@ def report_states():
         statement=text(
             "SELECT d.name, o.name, o.state, o.message FROM datasets AS d LEFT OUTER JOIN "
             "dataset_operations AS o ON o.dataset_ref = d.id"
-        )
-    )
+        ),
+        execution_options={"stream_results": True},
+    ).yield_per(1000)
```

Comment:
It would probably be good to use a symbolic constant for the `1000`, since I think it appears in a few places, now, and, at some point, we might want to raise it.

Reply:
I considered that; along with rewriting my literal SQL expression in SQLAlchemy, but I didn't. I've played with several chunk values now in different places; so, perhaps, although I feel like it's a little more awkward consolidating across cache manager and the report generator. I suppose it could be in `Database.__init__.py`...